hazelcast/src/main/java/com/hazelcast/flakeidgen/impl/FlakeIdGeneratorProxy.java

```diff
@@ -226,10 +227,18 @@ int getNodeId(long nanoTime) {
 
                 // we ignore possible double initialization
                 this.nodeId = nodeId;
+                nextNodeIdUpdate = nanoTime + NODE_ID_UPDATE_INTERVAL_NS;
                 if (logger.isFineEnabled()) {
                     logger.fine("Node ID assigned to '" + name + "': " + nodeId);
                 }
             }
+        } else if (nodeId == NODE_ID_NOT_YET_SET) {
+            // There is a race between the IF branch above and the local nodeId assignment just above the branch.
+            // It could be we didn't take the branch yet the local nodeId variable still contains the
```

Comment:
Minor: did you want to say "It could be that we did take the branch, yet the local nodeId variable is set to NODE_ID_NOT_YET_SET"? I must admit, I know what you did with the else branch but I don't understand the comment :)